### PR TITLE
Sync recent changes

### PR DESCRIPTION
**Pull request by PuruVJ**
## Description
- Remove smtp-client module and file (Unnecessary. Correct me if its needed, will add it back)
- Modify the tedious `format(new Date(tweet.created_at), 'MMM d, yyyy')` to be as simple as `formatTimestamp(tweets.created_at)`
- Modify the utils test to account for this new change. All tests passing

## Attach Screenshot
image

### CODE REVIEW

**Comment by PuruVJ** [comment]
Can somebody help me all the above commits tagging along?

I don't know how to get rid of these

**Comment by johnjacobkenny** [comment]
> Can somebody help me all the above commits tagging along?
> 
> I don't know how to get rid of these

Lol

**Comment by PuruVJ** [comment]
What do you think Kenny? Looks good enough?

**Review by johnjacobkenny — APPROVED** [review]
LGTM. @Mira-Alf please review.

**Comment by Mira-Alf** [comment]
Looks good to me. There is twitter-puppeteer/index.js and Twitter/scraping/index. Js. Which one is being used??

**Review by Mira-Alf — APPROVED** [review]

**Comment by PuruVJ** [comment]
@Mira-Alf I've set up the scraping one as the default in my upcoming commit.

**Comment by PuruVJ** [comment]
Right now its twitter-puppeteer

**Comment by Mira-Alf** [comment]
Either way we ll need to get rid of one. We can discuss it during the evening call. Also down the road we don't need the puppeteer flag because the application should do scraping only if tweet is older than 7 days.

**Comment by johnjacobkenny** [comment]
> @Mira-Alf I've set up the scraping one as the default in my upcoming commit.

Will wait before merging

**Comment by PuruVJ** [comment]
@johnjacobkenny Now you can merge it.

File: twindle-cli/twitter/conversation.json
Comment (johnjacobkenny) [inline twindle-cli/twitter/conversation.json]: Is this file needed?

File: twindle-cli/twitter/test/api.test.js
Comment (johnjacobkenny) [inline twindle-cli/twitter/test/api.test.js:15]: Any idea what is going wrong?

File: twindle-cli/twitter/transformations/tweets-array-endpoint.js
Comment (johnjacobkenny) [inline twindle-cli/twitter/transformations/tweets-array-endpoint.js:1]: What are the changes made in this file? Should @Mira-Alf take a look?

File: twindle-cli/twitter/types.d.ts
Comment (johnjacobkenny) [inline twindle-cli/twitter/types.d.ts:58]: If it is possible, you may move the type definitions into a `types` folder?

**Review by johnjacobkenny — APPROVED** [review]
I like the typing support. Please take a look at the comments. Nothing too big tho.

File: twindle-cli/twitter/test/api.test.js
Comment (PuruVJ) [reply twindle-cli/twitter/test/api.test.js:15]: No idea. It's saying that getTweetByID is not throwing the error when token is missing, but it is, when I manually tested it.

Maybe trying to resolve it with an `await` will fix it. I'll have to see how jest works.

File: twindle-cli/twitter/transformations/tweets-array-endpoint.js
Comment (PuruVJ) [reply twindle-cli/twitter/transformations/tweets-array-endpoint.js:1]: I didn't even touch it. Why is it here, I have no idea. lemme see

Comment (PuruVJ) [reply twindle-cli/twitter/transformations/tweets-array-endpoint.js:1]: That's weird. I have all that code right here. I'll push again

Comment (PuruVJ) [reply twindle-cli/twitter/transformations/tweets-array-endpoint.js:1]: Added back

File: twindle-cli/twitter/types.d.ts
Comment (PuruVJ) [reply twindle-cli/twitter/types.d.ts:58]: Ok. But we would need to keep the types.js in the root at least, so it can be used globally

**Comment by PuruVJ** [comment]
If you can merge, I'll start looking into how to resolve that Jest error about token.

File: twindle-cli/twitter/types.d.ts
Comment (proful) [reply twindle-cli/twitter/types.d.ts]: Typescript 😂

Comment (PuruVJ) [reply twindle-cli/twitter/types.d.ts]: @proful Not really

These are here to assist the coding experience. No compilers or extensions needed. Anyone with VScode and Atom will benefit, out of box.

**Comment by proful** [comment]
Please do not go wild and add typescript everywhere.

Typescript is not beginner friendly.

**Comment by johnjacobkenny** [comment]
@PuruVJ I'm not sure if you committed the typings folder

File: twindle-cli/twitter/types.d.ts
Comment (proful) [reply twindle-cli/twitter/types.d.ts]: This is fine but be careful of over usage.

**Comment by PuruVJ** [comment]
No one is adding typescript wildly anywhere. We're editing only in JS files only. Only those familiar with TS files need to concern about it.

No change for beginners

**Comment by johnjacobkenny** [comment]
> Please do not go wild and add typescript everywhere.
> 
> Typescript is not beginner friendly.

@proful I think in this case it may be transparent to anyone using it, since he has added a few typings files which is silently there.

**Comment by proful** [comment]
That's fine. Just sharing my thoughts for future.

**Comment by johnjacobkenny** [comment]
> @PuruVJ I'm not sure if you committed the typings folder

@PuruVJ will merge after your confirmation

**Comment by PuruVJ** [comment]
Committed

**Comment by PuruVJ** [comment]
Issue with keeping it in typings folder.

**Comment by johnjacobkenny** [comment]
Could you now merge into main with the changes? After that you can look at the tests

**Pull request merged**
